docs/design/integrate-with-cc-tools/README.md

```diff
@@ -0,0 +1,300 @@
+
+The integration between FPC and cc-tools needs to occur at two distinct levels:
+
+Chaincode Level: The chaincode itself will be written using cc-tools to simplify its complexity and must be compatible with the privacy features of FPC. This requires modifications to the chaincode lifecycle, ensuring that it can be deployed securely using FPC while retaining the flexibility of cc-tools for development.
+Note: Adhering to FPC's security requirements adds more complexity to the chaincode deployment process than with standard fabric.
+
+Client-Side Level: The client that communicates with the peers must be adapted to handle the secure interaction between the FPC chaincode and the Fabric network. This includes managing secure communication with the TEE and ensuring that FPC's privacy guarantees are upheld during transaction processing.
+
+Addressing these challenges is critical to enabling the seamless development and deployment of private chaincode on Hyperledger Fabric, ensuring both security and usability.
+Note: the following list comprises the functionality supported by cc-tools but not yet by FPC:
+
+* Transient data
+* Events
+* Private Data
+
+## Proposed Solution
+
+### On The Chaincode Level
+
+In developing the chaincode for the integration of FPC and cc-tools, there are specific requirements to ensure that FPC's privacy and security features are properly leveraged. First, both FPC and cc-tools wrap the [shim.ChaincodeStub](https://github.com/hyperledger/fabric-chaincode-go/blob/acf92c9984733fb937fba943fbf7397d54368751/shim/interfaces.go#L28) interface from Hyperledger Fabric, but they must be used in a specific order for it to function. This order is determined by a few factors:
+
+1. Stub interface order:
+
+    cc-tools is a package that provides a relational-like framework for programming fabric chaincodes and it translates every code to a normal fabric chaincode at the end. cc-tools is wrapping the [shim.ChaincodeStub](https://github.com/hyperledger/fabric-chaincode-go/blob/acf92c9984733fb937fba943fbf7397d54368751/shim/interfaces.go#L28) interface from Hyperledger Fabric using [stubWrapper](https://github.com/hyperledger-labs/cc-tools/blob/995dfb2a16decae95a9dbf05424819a1df19abee/stubwrapper/stubWrapper.go#L12) and if you look for example at the [PutState](https://github.com/hyperledger-labs/cc-tools/blob/995dfb2a16decae95a9dbf05424819a1df19abee/stubwrapper/stubWrapper.go#L18) function you notice it only does some in-memory operations and it calls another `sw.Stub.PutState` from the stub passed to it (till now it was always the [stub for standard fabric](https://github.com/hyperledger/fabric-chaincode-go/blob/main/shim/stub.go)).
+
+    ```
```

Comment:
```suggestion
    ```go
```

Reply:
I think the code formatting looks much better already! If you set the code block type to `go`, we even get color :)

Reply:
Also it seems that the code blocks still have different formats, in particular the spacing.

Reply:
> I think the code formatting looks much better already! If you set the code block type to go, we even get color :)
Done 
> Also it seems that the code blocks still have different formats, in particular the spacing.
Can you point to an example?

Reply:
the ones in the user experience section look different.